src/org/traccar/protocol/SviasProtocolDecoder.java

```diff
@@ -0,0 +1,150 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.helper.PatternBuilder;
+
+import java.net.SocketAddress;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.regex.Pattern;
+import org.traccar.DeviceSession;
+import org.traccar.helper.Parser;
+import org.traccar.helper.UnitsConverter;
+import org.traccar.model.Position;
+
+public class SviasProtocolDecoder extends BaseProtocolDecoder {
+
+    private static final String MSG_KEEPALIVE = "@";
+
+    public SviasProtocolDecoder(SviasProtocol protocol) {
+        super(protocol);
+    }
+
+    private static final Pattern PATTERN = new PatternBuilder()
+            .text("[")       // delimiter init
+            .number("(dddd),") // version hardware
+            .number("(dddd),") // version software
+            .number("(d+),") // counter
+            .number("(d+),") // imei
+            .any()            // model or hourmeter
+            .number("(d+),") // date (yyyymmdd)
+            .number("(d+),") // time (hhmmss)
+            .number("(-?d+),") // longitude
+            .number("(-?d+),") // latitude
+            .number("(d+),") // speed
+            .number("(d+),") // course
+            .number("(d+),") // odometer
+            .number("(d+),") // input
+            .number("(d+),") // output / status
+            .number("(d+),") // flag pack input 1
+            .number("(d+),") // flag pack input 2
+            .number("(d+),") // main power voltage
+            .number("(d+),") // percentual power internal battery
+            .number("(d+),") // RSSID
+            .any()
+            .compile();
+
+    private double convertCoordinates(long v) {
+        return Double.valueOf(((float) ((((float) v / 1.0E7F)
+                - ((int) (v / 10000000L))) * 1.6666666666666667D)) + ((int) (v / 10000000L)));
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg)
+            throws Exception {
+
+        String sentence = (String) msg;
+
+        if (!sentence.contains(":")) {
+
+            Parser parser = new Parser(PATTERN, (String) sentence);
+            if (!parser.matches()) {
+                return null;
+            }
+
+            Position position = new Position(getProtocolName());
+
+            String versionHard = parser.next();

```

Comment:
Remove any unused variables. There is no need to capture a group if you don't plan to use it.

Reply:
Done!